dialer.go

```diff
@@ -81,11 +85,20 @@ func (d *dialer) DialStream() (Stream, error) {
 		idsExhausted = true
 		d.id = 0
 	}
+	idled := false
+	if d.idleInterval > 0 {
+		now := time.Now()
+		idled = now.Sub(d.lastDialed) > d.idleInterval
+		if idled {
```

Comment:
This keeps us from using a single TCP connection for too long, which is important both because that messes with BBR statistics and also because it's probably better from a blocking-resistance perspective to not have a super long-lived connection.